src/java.base/share/classes/java/lang/Runtime.java

```diff
@@ -83,10 +83,9 @@ private Runtime() {}
      * until they finish.  Once this is done the virtual machine
      * {@linkplain #halt halts}.
      *
-     * <p> If this method is invoked after all shutdown hooks have already
-     * been run and the status is nonzero then this method halts the
-     * virtual machine with the given status code. Otherwise, this method
-     * blocks indefinitely.
+     * <p> Invocations of this method block indefinitely. It is therefore
+     * inadvisable to invoke this method from a shutdown hook as it will
+     * cause deadlock.
```

Comment:
This is inaccurate. The method only blocks indefinitely when shutdown has already commenced.

Reply:
If a shutdown hook is running, then shutdown has started, right? This new wording isn’t really a change, it’s the current behavior (the “otherwise” portion). But the wording is meant to be more accurate for the case of running exit from shutdown hooks. The change is the removal of the first case, which is what the code portion of the PR removes.

Reply:
The first paragraph of the method description already makes it clear that the method never returns normally. One option for the third paragraph is to just remove it, another is to replace it with wording that specifies that the first thread to call exit is the winner and the exit code provided by other threads that attempt to exit around the same time (or while exit hooks execute) will be ignored. I think there is merit with adding a warning that a shutdown hook invoking exit will deadlock.

Reply:
Thanks for the feedback, they are all good points. I opted to make the behavior with multiple threads more clear, see [8e7d527](https://github.com/openjdk/jdk/pull/9351/commits/8e7d527a182933818bd2d7f0f1b799dac52663be).